pylint/checkers/refactoring/refactoring_checker.py

```diff
@@ -1393,7 +1393,18 @@ def _check_unnecessary_comprehension(self, node):
         else:
             return
         if expr_list == target_list != []:
-            self.add_message("unnecessary-comprehension", node=node)
+            if isinstance(node.parent, astroid.DictComp):
+                func = "dict"
+            elif isinstance(node.parent, astroid.ListComp):
+                func = "list"
+            else:
+                func = "set"
```

Comment:
Probably a bit over cautious, but I would prefer an explicit `elif ... SetComp):` together with `else: return`. That way its future prove in case something changes.

Reply:
fixed here:
https://github.com/PyCQA/pylint/pull/4500/files#diff-af1d215b95c8b3ac59218ba588186da4aefd936eef4732518196ea309c3cc788R1423